### PR TITLE
VMHostIScsiHba and VMHostIScsiHbaTarget DSC Resources

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Introduced VMHostIScsiHbaBaseDSC class.
- Added VMHostIScsiHba DSC Resource.
- Added VMHostIScsiHbaTarget DSC Resource.
- Added Unit Tests for VMHostIScsiHbaBaseDSC class.
- Added Unit Tests for VMHostIScsiHba DSC Resource.
- Added Unit Tests for VMHostIScsiHbaTarget DSC Resource.
- Added Integration Tests for VMHostIScsiHba DSC Resource.
- Added Integration Tests for VMHostIScsiHbaTarget DSC Resource.
- Added Documentation and example Configurations for VMHostIScsiHba DSC Resource.
- Added Documentation and example Configurations for VMHostIScsiHbaTarget DSC Resource.

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Example usage of VMHostIScsiHba DSC Resource:

```
VMHostIScsiHba VMHostIScsiHba {
    Server = $Server
    Credential = $Credential
    VMHostName = $VMHostName
    Name = $IScsiHbaName
    ChapType = 'Required'
    ChapName = 'AdminOne'
    ChapPassword = 'AdminPasswordOne'
    MutualChapEnabled = $true
    MutualChapName = 'AdminTwo'
    MutualChapPassword = 'AdminPasswordTwo'
}
```

**Comment by SimeonGerginov** [comment]
Example usage of VMHostIScsiHbaTarget DSC Resource:

```
VMHostIScsiHbaTarget VMHostIScsiHbaTarget {
    Server = $Server
    Credential = $Credential
    VMHostName = $VMHostName
    Address = '10.23.84.73'
    Port = 3260
    IScsiHbaName = $IScsiHbaName
    TargetType = 'Send'
    Ensure = 'Present'
    InheritChap = $false
    ChapType = 'Required'
    ChapName = 'AdminOne'
    ChapPassword = 'AdminPasswordOne'
    InheritMutualChap = $false
    MutualChapEnabled = $true
    MutualChapName = 'AdminTwo'
    MutualChapPassword = 'AdminPasswordTwo'
}
```

File: Source/VMware.vSphereDSC/DSCResources/VMHostIScsiHba.ps1 [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    #>
+    [void] ConfigureIScsiHbaChap($iScsiHba) {
+        $setVMHostHbaParams = @{
+            IScsiHba = $iScsiHba
+            Confirm = $false
+            ErrorAction = 'Stop'
+            Verbose = $false
+        }
+
+        $this.PopulateCmdletParametersWithCHAPSettings($setVMHostHbaParams, $null, $null)
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/DSCResources/VMHostIScsiHba.ps1]: An overload may be added without last two parameters

Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/DSCResources/VMHostIScsiHba.ps1]: Method overload was added to ShouldModifyCHAPSettings() and PopulateCmdletParametersWithCHAPSettings().

**Pull request merged**